lib/checkbufferoverrun.cpp

```diff
@@ -1015,6 +1015,10 @@ void CheckBufferOverrun::objectIndex()
                     if (var->valueType()->pointer > obj->valueType()->pointer)
                         continue;
                 }
+                if ((obj->isCast() || (mTokenizer->isCPP() && isCPPCast(obj))) && obj->valueType()) { // allow cast to a different type
```

Comment:
Will this work when its assigned to a variable?

```cpp
uint32_t f(uint32_t u) {
    uint8_t* p = ((uint8_t*)&u);
    return p[3];
}
```

It seems like it would be better to check if the `valueType` is different and then check for overrun index. If the type size is not known for `var` then we should skip.

Reply:
That gave a FP, fixed now.